autograd.cpp

```diff
@@ -311,6 +297,13 @@ void OptimizerImpl::zero_grad() {
   }
 }
 
+void OptimizerImpl::attach(Container model, bool clearState) {
```

Comment:
`set_model()`? The `clearState` parameter is a bit confusing IMHO, you could require people to just call `init_state()` instead.

Reply:
The reason I have this is that I want to be able to save the optimizer and its model seperately, so functions that archive them must then attach the model afterwards. Sure, I can rename it to set_model.

Reply:
Got it, just wanted to highlight that if you see an `attach` call in code somewhere it's not clear what the second parameter (true/false) would mean and calling `init_state()` if needed is more expressive.